src/dparse/formatter.d

```diff
@@ -2678,7 +2678,8 @@ class Formatter(Sink)
         {
             foreach (contract; functionContracts)
                 format(contract);
-            put("do");
+            if (specifiedFunctionBody.hasDo)
+                put("do");
```

Comment:
Looks like the line is not covered though ? And neither is `format(contract)`

Reply:
Fixed.